### PR TITLE
Move configuration information out of example usage page

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Part of https://github.com/apache/datafusion/issues/11172

## Rationale for this change

The current example usage page  https://datafusion.apache.org/user-guide/example-usage.html contains a mix of getting started examples  as well as more advanced options such as patching in new datafusion versions

Screenshot 2024-07-06 at 6 50 31 AM

I think this would be easier for people to find if they were split into two separate pages

## What changes are included in this PR?

Move some of the more advanced usage examples to a new page called "Crate Configuration"

Screenshot 2024-07-06 at 6 52 18 AM

Screenshot 2024-07-06 at 6 53 49 AM

## Are these changes tested?
I manually looked at them

## Are there any user-facing changes?

No, doc changes only

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -41,13 +41,16 @@ DataFusion offers SQL and Dataframe APIs, excellent
 CSV, Parquet, JSON, and Avro, extensive customization, and a great
 community.
 
-To get started with examples, see the `example usage`_ section of the user guide and the `datafusion-examples`_ directory.
+To get started, see
```
Comment (alamb) [inline docs/source/index.rst:44]: I also added a link to the library user guide as well as making this into a bullet list so it is easier to see what options are available

File: docs/source/user-guide/crate-configuration.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+Based on the instruction set architecture you are building on you will want to configure the `target-cpu` as well, ideally
+with `native` or at least `avx2`.
+
+```shell
+RUSTFLAGS='-C target-cpu=native' cargo run --release
+```
+
+## Enable backtraces
+
+By default Datafusion returns errors as a plain message. There is option to enable more verbose details about the error,
```
Comment (jonahgao) [inline docs/source/user-guide/crate-configuration.md:92]:
```suggestion
By default DataFusion returns errors as a plain message. There is option to enable more verbose details about the error,
```

File: docs/source/user-guide/crate-configuration.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+like error backtrace. To enable a backtrace you need to add Datafusion `backtrace` feature to your `Cargo.toml` file:
```
Comment (jonahgao) [inline docs/source/user-guide/crate-configuration.md:93]:
```suggestion
like error backtrace. To enable a backtrace you need to add DataFusion `backtrace` feature to your `Cargo.toml` file:
```

**Review by jonahgao — APPROVED** [review]
Looks good to me!

File: docs/source/user-guide/crate-configuration.md (hunk 3 above)
Comment (alamb) [reply docs/source/user-guide/crate-configuration.md:93]: Done in https://github.com/apache/datafusion/pull/11439

**Comment by alamb** [comment]
Thanks for the review @jonahgao and merge @comphead  🙏

**Pull request merged**